providers/config/providerConfig.go

```diff
@@ -38,6 +49,38 @@ func LoadProviderConfigs(fname string) (map[string]map[string]string, error) {
 	return results, nil
 }
 
+func isExecutable(filename string) bool {
+	if stat, statErr := os.Stat(filename); statErr == nil {
+		if mode := stat.Mode(); mode&0111 == 0111 {
+			return true
+		}
+	}
+	return false
+}
+
+func readCredsFile(filename string) ([]byte, error) {
+	dat, err := utfutil.ReadFile(filename, utfutil.POSIX)
+	if err != nil {
+		// no creds file is ok. Bind requires nothing for example. Individual providers will error if things not found.
+		if os.IsNotExist(err) {
+			fmt.Printf("INFO: Config file %q does not exist. Skipping.\n", filename)
+			return []byte{}, nil
+		}
+		return nil, fmt.Errorf("failed reading provider credentials file %v: %v", filename, err)
+	}
+	return dat, nil
+}
+
+func executeCredsFile(filename string) ([]byte, error) {
+	cmd := filename
+	if !strings.HasPrefix(filename, "/") {
+		// if the path doesn't start with `/` make sure we aren't relying on $PATH.
+		cmd = "./" + filename
```

Comment:
Suggestion: `cmd = filepath.Join(".", filename)` 

(this way it will work on Windows)

Reply:
If i do this, it relies on $PATH:

```
$ dnscontrol check-creds --creds=creds.sh powerdns POWERDNS 
failed GetZone lpc("creds.sh"): exec: "creds.sh": executable file not found in $PATH
```

Reply:
Dang it!  The filepath.Join() function tries to be smart and clean up afterwords.  Ugh.  How about:

    strings.Join(".", filepath.Separator, filename)

Reply:
Yes, this works like expected.